src/execution_plan/optimizations/reduce_traversal.c

```diff
@@ -65,9 +64,20 @@ void reduceTraversal(ExecutionPlan *plan) {
 		   AlgebraicExpression_OperandCount(ae) == 1 &&
 		   AlgebraicExpression_DiagonalOperand(ae, 0)) continue;
 
-		/* Search to see if dest is already resolved */
-		if(!ExecutionPlan_LocateOpResolvingAlias(op->children[0],
-												 AlgebraicExpression_Destination(ae))) continue;
+		// Collect variables bound before this op.
+		rax *bound_vars = raxNew();
+		for(int i = 0; i < op->childCount; i ++) {
+			ExecutionPlan_BoundVariables(op->children[i], bound_vars);
+		}
+
+		const char *src = AlgebraicExpression_Source(ae);
+		const char *dest = AlgebraicExpression_Destination(ae);
+		if(raxFind(bound_vars, (unsigned char *)src, strlen(src)) == raxNotFound ||
```

Comment:
I don't think we should look for `src` within `bound_vars` the fact that there's a traversal (`op`) should guarantee `src` is bounded.

Reply:
That's fair, removed!